### ISSUE TITLE
tb-node microservice v2.5 fails to start due to missing dependency annotations or config

### ISSUE THREAD

**Issue by sibradzic** [thingsboard/thingsboard#2838]
Hello and thank you guys for this great project!

I'm thoroughly following the k8s micro-service setup guide https://github.com/thingsboard/thingsboard/blob/master/k8s/README.md, and i could deploy and end up with a working TB system when using release 2.4 or SNAPSHOT-2.5 (before release 2.5 was branched).
Now with release-2.5 (self-built `release-2.5` branch images) or master (`latest` images), micro-services still deploy fine, but **tb-node** are falling into CrashLoopBackOff, producing the following log output (last lines that really matter, full log attached below):
```
2020-05-26 09:15:13,296 [main] INFO  o.t.s.q.d.DefaultTbServiceInfoProvider - Current Service ID: env-node-5664cffcc-vdcbg
2020-05-26 09:15:14,411 [main] WARN  o.s.b.w.s.c.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'remoteJsInvokeService': Unsatisfied dependency expressed through field 'requestTemplate'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.thingsboard.server.queue.TbQueueRequestTemplate<org.thingsboard.server.queue.common.TbProtoJsQueueMsg<org.thingsboard.server.gen.js.JsInvokeProtos$RemoteJsRequest>, org.thingsboard.server.queue.common.TbProtoQueueMsg<org.thingsboard.server.gen.js.JsInvokeProtos$RemoteJsResponse>>' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.springframework.beans.factory.annotation.Qualifier(value=RemoteJsInvokeService)}
2020-05-26 09:15:14,419 [nosql-dispatcher-1-thread-1] INFO  o.t.s.d.u.AbstractBufferedRateExecutor - Buffered rate executor thread stopped
2020-05-26 09:15:14,419 [nosql-dispatcher-1-thread-2] INFO  o.t.s.d.u.AbstractBufferedRateExecutor - Buffered rate executor thread stopped
2020-05-26 09:15:16,464 [main] INFO  o.a.catalina.core.StandardService - Stopping service [Tomcat]
2020-05-26 09:15:16,536 [main] INFO  o.s.b.a.l.ConditionEvaluationReportLoggingListener - 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2020-05-26 09:15:17,194 [main] ERROR o.s.b.d.LoggingFailureAnalysisReporter - 

***************************
APPLICATION FAILED TO START
***************************

Description:

Field requestTemplate in org.thingsboard.server.service.script.RemoteJsInvokeService required a bean of type 'org.thingsboard.server.queue.TbQueueRequestTemplate' that could not be found.

The injection point has the following annotations:
	- @org.springframework.beans.factory.annotation.Autowired(required=true)
	- @org.springframework.beans.factory.annotation.Qualifier(value=RemoteJsInvokeService)

The following candidates were found but could not be injected:
	- Bean method 'createRemoteJsRequestTemplate' in 'AwsSqsMonolithQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='aws-sqs' && '${service.type:null}'=='monolith'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'AwsSqsTbCoreQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='aws-sqs' && '${service.type:null}'=='tb-core'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'AwsSqsTbRuleEngineQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='aws-sqs' && '${service.type:null}'=='tb-rule-engine'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'KafkaMonolithQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='kafka' && '${service.type:null}'=='monolith'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'KafkaTbCoreQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='kafka' && '${service.type:null}'=='tb-core'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'KafkaTbRuleEngineQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='kafka' && '${service.type:null}'=='tb-rule-engine'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'PubSubMonolithQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='pubsub' && '${service.type:null}'=='monolith'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'PubSubTbCoreQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='pubsub' && '${service.type:null}'=='tb-core'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'PubSubTbRuleEngineQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='pubsub' && '${service.type:null}'=='tb-rule-engine'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'RabbitMqMonolithQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='rabbitmq' && '${service.type:null}'=='monolith'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'RabbitMqTbCoreQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='rabbitmq' && '${service.type:null}'=='tb-core'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'RabbitMqTbRuleEngineQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='rabbitmq' && '${service.type:null}'=='tb-rule-engine'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'ServiceBusMonolithQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='service-bus' && '${service.type:null}'=='monolith'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'ServiceBusTbCoreQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='service-bus' && '${service.type:null}'=='tb-core'}) resulted in false
	- Bean method 'createRemoteJsRequestTemplate' in 'ServiceBusTbRuleEngineQueueFactory' not loaded because @ConditionalOnExpression (#{'${queue.type:null}'=='service-bus' && '${service.type:null}'=='tb-rule-engine'}) resulted in false

Action:

Consider revisiting the entries above or defining a bean of type 'org.thingsboard.server.queue.TbQueueRequestTemplate' in your configuration.
```
Following the dependency chain, something may be off with:
```
No qualifying bean of type
  org.thingsboard.server.queue.TbQueueRequestTemplate
    <
      org.thingsboard.server.queue.common.TbProtoJsQueueMsg <org.thingsboard.server.gen.js.JsInvokeProtos$RemoteJsRequest>,
      org.thingsboard.server.queue.common.TbProtoQueueMsg   <org.thingsboard.server.gen.js.JsInvokeProtos$RemoteJsResponse>
    >
expected at least 1 bean which qualifies as autowire candidate. Dependency annotations:
  {@org.springframework.beans.factory.annotation.Autowired(required=true)}
```
which points to https://github.com/thingsboard/thingsboard/blob/release-2.5/application/src/main/java/org/thingsboard/server/service/script/RemoteJsInvokeService.java#L83-L84
Could it be that some `@Quallifier` annotations are missing there, or perhaps there is some config missing to properly initialize this "remote JS" service?

Note that everything works just fine when running the stand-alone variant. But as *RemoteJsInvokeService* is only activated conditionally via https://github.com/thingsboard/thingsboard/blob/release-2.5/application/src/main/java/org/thingsboard/server/service/script/RemoteJsInvokeService.java#L45, perhaps this does not go through same tests...

Thanks again and please let me know if there something I could try to fix this?

**Comment by sibradzic** [comment]
full log at: https://pastebin.com/AQzGAABK

**Comment by sibradzic** [comment]
In addition to tb-node, all transport services are also failing in a similar manner:
```
2020-05-26 10:07:35,830 [main] INFO  o.a.kafka.common.utils.AppInfoParser - Kafka startTimeMs: 1590487655830
2020-05-26 10:07:35,865 [main] WARN  o.s.b.w.s.c.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'defaultTransportService' defined in URL [jar:file:/usr/share/tb-http-transport/bin/tb-http-transport.jar!/BOOT-INF/lib/transport-api-2.5.0.jar!/org/thingsboard/server/common/transport/service/DefaultTransportService.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'tbCoreQueueProducerProvider' defined in URL [jar:file:/usr/share/tb-http-transport/bin/tb-http-transport.jar!/BOOT-INF/lib/queue-2.5.0.jar!/org/thingsboard/server/queue/provider/TbCoreQueueProducerProvider.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'kafkaMonolithQueueFactory' defined in URL [jar:file:/usr/share/tb-http-transport/bin/tb-http-transport.jar!/BOOT-INF/lib/queue-2.5.0.jar!/org/thingsboard/server/queue/provider/KafkaMonolithQueueFactory.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'hashPartitionService' defined in URL [jar:file:/usr/share/tb-http-transport/bin/tb-http-transport.jar!/BOOT-INF/lib/queue-2.5.0.jar!/org/thingsboard/server/queue/discovery/HashPartitionService.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.thingsboard.server.queue.discovery.TenantRoutingInfoService' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
2020-05-26 10:07:35,896 [main] INFO  o.a.catalina.core.StandardService - Stopping service [Tomcat]
2020-05-26 10:07:35,917 [main] INFO  o.s.b.a.l.ConditionEvaluationReportLoggingListener - 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2020-05-26 10:07:36,224 [main] ERROR o.s.b.d.LoggingFailureAnalysisReporter - 

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of constructor in org.thingsboard.server.queue.discovery.HashPartitionService required a bean of type 'org.thingsboard.server.queue.discovery.TenantRoutingInfoService' that could not be found.


Action:

Consider defining a bean of type 'org.thingsboard.server.queue.discovery.TenantRoutingInfoService' in your configuration.
```

**Comment by sibradzic** [comment]
OK, after reading some code, in case of **tb-node**, this looks like missing `TB_QUEUE_TYPE` env var in the container / deployment definition, after adding it the service starts without complaining. It does not fix transport services though, something else may be missing there...

**Comment by sibradzic** [comment]
OK, after removing
```
        - name: TB_SERVICE_TYPE
          value: "monolith"
```
from deployment definitions of transport services, all services are now starting...

https://github.com/thingsboard/thingsboard/blob/release-2.5/k8s/thingsboard.yml#L176-L177
https://github.com/thingsboard/thingsboard/blob/release-2.5/k8s/thingsboard.yml#L249-L250
https://github.com/thingsboard/thingsboard/blob/release-2.5/k8s/thingsboard.yml#L322-L323

**Comment by ashvayka** [comment]
Hi @sibradzic 

I believe what is really missing is the TB_QUEUE_TYPE that should be set to "kafka". The monolith is ok. We will update this asap.

**Comment by ashvayka** [comment]
@vzikratyi-tb please help to fix this in develop/2.5.1 and master branches of TB CE.

**Comment by sibradzic** [comment]
> Hi @sibradzic
> 
> I believe what is really missing is the TB_QUEUE_TYPE that should be set to "kafka". The monolith is ok. We will update this asap.

Yes, right, that is what I meant. Obviously, with the rest of the services built around kafka,
```
        - name: TB_QUEUE_TYPE
          value: kafka
```
is the only thing that makes sense.
